exporter.py

```diff
@@ -0,0 +1,100 @@
+from optparse import OptionParser
+from prometheus_client import REGISTRY, start_http_server
+
+import importlib
+import logging
+import os
+import time
+import urllib3
+import modules.Configuration as config
+
+# disable ssl warning
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+# init logger once
+logger = logging.getLogger('esxi-exporter')
+parser = OptionParser()
+parser.add_option("-d", "--debug", dest="debug", help="enable debug output",
+                  action="store_true")
+parser.add_option("-v", "--info", dest="info", help="enable info output",
+                  action="store_true")
+(options, args) = parser.parse_args()
+if options.debug:
+    logger.setLevel(logging.DEBUG)
+elif options.info:
+    logger.setLevel(logging.INFO)
+else:
+    logger.setLevel(logging.WARNING)
+ch = logging.StreamHandler()
+formatter = logging.Formatter('[%(asctime)s] [%(levelname)s] %(message)s')
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+
+def run_prometheus_server(port: int) -> None:
+    """
+    Starts a prometheus server which will stay running
+    :param port: The port the server should listen on
+    """
+
+    # startup prometheus server
+    logger.debug("starting http server...")
+    start_http_server(port)
+
+    for collector in get_collectors():
+        logger.debug(
+            f"registering collector: {collector.__name__.split('.')[-1]}")
+        REGISTRY.register(collector())

```

Comment:
Why using `collector()` as a function here?

Reply:
because it is a list of classes and it needs to be initialized. if wanted I can put initialization in factory